packages/ethereum-light-client/src/verify.rs

```diff
@@ -0,0 +1,396 @@
+            get_subtree_index(NEXT_SYNC_COMMITTEE_INDEX),
+            update.attested_header.beacon.state_root,
+        )
+        .map_err(|e| EthereumIBCError::ValidateNextSyncCommitteeFailed(Box::new(e)))?;
+    }
+
+    // Verify sync committee aggregate signature
+    let sync_committee = if signature_period == stored_period {
+        trusted_consensus_state
+            .current_sync_committee()
+            .ok_or(EthereumIBCError::ExpectedCurrentSyncCommittee)?
+    } else {
+        trusted_consensus_state
+            .next_sync_committee()
+            .ok_or(EthereumIBCError::ExpectedNextSyncCommittee)?
+    };
+
+    // It's not mandatory for all of the members of the sync committee to participate. So we are extracting the
+    // public keys of the ones who participated.
+    let participant_pubkeys = update
+        .sync_aggregate
+        .sync_committee_bits
+        .iter()
+        .flat_map(|byte| (0..8).rev().map(move |i| (byte & (1 << i)) != 0))
+        .zip(sync_committee.pubkeys.0.iter())
+        .filter_map(|(included, pubkey)| included.then_some(pubkey))
+        .collect::<Vec<_>>();
+
+    let fork_version_slot = std::cmp::max(update.signature_slot, 1) - 1;
+    let fork_version = compute_fork_version(
+        &client_state.fork_parameters,
+        compute_epoch_at_slot(client_state.slots_per_epoch, fork_version_slot),
+    );
+
+    let domain = compute_domain(
+        DomainType::SYNC_COMMITTEE,
+        Some(fork_version),
+        Some(client_state.genesis_validators_root),
+        client_state.fork_parameters.genesis_fork_version.clone(),
+    );
+    let signing_root = compute_signing_root(&update.attested_header.beacon, domain);
+
+    bls_verifier
+        .fast_aggregate_verify(
+            participant_pubkeys,
+            signing_root,
+            update.sync_aggregate.sync_committee_signature,
+        )
+        .map_err(|err| EthereumIBCError::FastAggregateVerify(err.to_string()))?;
+
+    Ok(())
+}
+
+/// Validates a light client header.
+///
+/// [See in consensus-spec](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/light-client/sync-protocol.md#modified-is_valid_light_client_header)
+pub fn is_valid_light_client_header(
+    client_state: &ClientState,
+    header: &LightClientHeader,
+) -> Result<(), EthereumIBCError> {
+    let epoch = compute_epoch_at_slot(client_state.slots_per_epoch, header.beacon.slot);
+
+    if epoch < client_state.fork_parameters.deneb.epoch {
+        ensure(
+            header.execution.blob_gas_used == 0 && header.execution.excess_blob_gas == 0,
+            EthereumIBCError::MustBeDeneb,
+        )?;
+    }
+
+    ensure(
+        epoch >= client_state.fork_parameters.capella.epoch,
+        EthereumIBCError::InvalidChainVersion,
+    )?;
+
+    validate_merkle_branch(
+        get_lc_execution_root(client_state, header),
+        header.execution_branch.0.into(),
+        floorlog2(EXECUTION_PAYLOAD_INDEX),
```

Comment:
There is a constant for this `EXECUTION_BRANCH_SIZE` right?

Reply:
Good point. Will go through them and see if I can consolidate and get rid of floorlog2

Reply:
I just noticed that floorlog2 has some extra safety added in. Not sure if we want to have that?
```rust
/// Convenience function safely to call [`u64::ilog2`] and convert the result into a usize.
#[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
```

Reply:
I kept the floorlog2, but only using it inside the config.rs which needs to be cleaned up anyway. I think this is fine for now at least. The rest of the code does not use it.